vm/starnix/starnix.go

```diff
@@ -248,13 +248,28 @@ func (inst *instance) connectToAdb(timeout time.Duration) error {
 }
 
 func (inst *instance) restartAdbAsRoot() error {
-	err := inst.runCommand(
-		"adb",
-		"-s",
-		fmt.Sprintf("127.0.0.1:%d", inst.port),
-		"root",
-	)
-	return err
+	startTime := time.Now()
+	for {
+		vmimpl.SleepInterruptible(3 * time.Second)
+		if inst.debug {
+			log.Logf(0, "attempting to restart ADB with root access")
+		}
+		err := inst.runCommand(
+			"adb",
+			"-s",
+			fmt.Sprintf("127.0.0.1:%d", inst.port),
+			"root",
+		)
+		if err == nil {
+			return nil
+		}
+		if inst.debug {
+			log.Logf(0, "adb root failed")
```

Comment:
Nit: there's `log.Error()` that might be a better alternative here.

Reply:
If it's okay with you, I think log.Logf(1, ...) is a better fit here, since this is expected to fail a few times before succeeding and we probably don't want to print anything while retrying.

Reply:
Btw, sorry for dismissing the review :) I'm used to a gerrit workflow where commits should be amended on each review pass. Maybe it would have been better to add the changes to a new commit in the PR here?

Reply:
No worries!
That's how our project is configured -- all pushes (except for pure rebases) drop the "reviewed" state. I'll mark it reviewed before merging.

Regarding the log lines: I think you'd also want to prepend instance names to the messages like you did here: `log.Logf(0, "instance %s booted successfully", inst.name)`. Otherwise it will be hard to understand what's going on once there are more than 1 VM.

Reply:
Okay, thank you!

And thanks for the suggestion; added the instance name to the log lines.